cloudflare/resource_cloudflare_fallback_domain.go

```diff
@@ -18,6 +27,7 @@ func resourceCloudflareFallbackDomain() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: resourceCloudflareFallbackDomainImport,
 		},
+		CustomizeDiff: resourceCloudflareFallbackDomainDiff,
```

Comment:
we should avoid `CustomizeDiff` where we can as it is pretty riddled with edge cases for certain types like `TypeMap` and behaviours on computed fields. it's not really something we want more of.'

if we're using the `CustomizeDiff` here for including the default fallback domains in addition to custom values and avoid permadiffs, we could instead make end users explicitly provide them which feels like expected behaviour given we are customising the already existing resource with their own domains.

Reply:
a note in the documentation mentioning this for the end user seems like it would address the surprise without any additional work in the provider itself.

Reply:
Yeah. I started this by just updating Delete to reset the entries back to the default set, and then figured, well, if we already have to store the list internally and the existing docs all imply you're _adding_ domains, might as well support the currently documented behavior.  I also tried to point out this documentation problem in #1381, but you immediately closed that issue.

Reply:
And without `CustomizeDiff`, we'd be able to change the type to `TypeSet` instead, which is actually more appropriate, since the order doesn't matter.